src/App.js

```diff
@@ -174,9 +175,59 @@ const useStyles = makeStyles((theme) => ({
       display: "none",
     },
   },
+  noTorContainer: {
+    fontSize: 24,
+    marginTop: 24,
+    width: "100%",
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+  },
+  noTorImage: {
+    backgroundImage: `url(${noTor})`,
+    backgroundRepeat: "no-repeat",
+    backgroundSize: "contain",
+    width: 100,
+    height: 100,
+    marginTop: 24,
+  },
 }));
 
 function App() {
+  const [isBlockedIP, setIsBlockedIP] = useState(false);
+
+  const checkForTOR = async () => {
+    // if not on mainnet, no need to check for TOR
+    if (process.env.REACT_APP_CLUSTER !== "mainnet") {
+      return false;
+    } else {
```

Comment:
Maybe we can do this way:

```suggestion
    if (process.env.REACT_APP_CLUSTER === "mainnet") {
      ...check tor ips
    } 
```

Reply:
The else was not needed, the second if neither 🙌🏼